packages/x-date-pickers/src/internals/utils/date-utils.ts

```diff
@@ -53,21 +53,21 @@ export const findClosestEnabledDate = <TDate>({
     }
 
     if (forward) {
-      if (!shouldDisableDate(forward)) {
+      if (!isDateDisabled(forward)) {
         return forward;
       }
       forward = utils.addDays(forward, 1);
     }
 
     if (backward) {
-      if (!shouldDisableDate(backward)) {
+      if (!isDateDisabled(backward)) {
         return backward;
       }
       backward = utils.addDays(backward, -1);
     }
   }
 
-  return today;
+  return null;
```

Comment:
I agree it was a bad idea because we need to keep in mind that there is not necessarily an enabled date, and dealing with this edge case is not the responsibility of this function

For the second point I think it makes more sens to move to `null` because having `onChange` called with a disabled date sounds like a bug